src/Microsoft.Extensions.Configuration.Abstractions/IConfigurationBuilder.cs

```diff
@@ -26,11 +26,11 @@ public interface IConfigurationBuilder
         /// </summary>
         /// <param name="source">The configuration source to add.</param>
         /// <returns>The same <see cref="IConfigurationBuilder"/>.</returns>
-        IConfigurationBuilder Add(IConfigurationSource source);
+        new IConfigurationBuilder Add(IConfigurationSource source);
```

Comment:
Why not make the Sources property an `IList` instead of the `IConfigurationBuilder`? Seems cleaner.